### DISCUSSION TITLE
autopush

### DISCUSSION THREAD

**Discussion by rubiin** [twpayne/chezmoi#3795] [General]
allow setting author on autocommit and push

for example in git, we do this to set a different author
```
git commit -m "$commit_msg" --author="🤖 automated<cron@job>"
```
I was looking for something similar on autocommit and push

If you could direct me , maybe i could come up with a little pr :)

**Comment by twpayne** [comment]
You can already do this with a `run_onchange_` script:

```sh
#!/bin/sh

git -S $CHEZMOI_WORKING_TREE config --local user.name "🤖 automated"
git -S $CHEZMOI_WORKING_TREE config --local user.email "cron@job"
```

**Reply by rubiin** [reply]
letme try

**Reply by rubiin** [reply]
![image](https://github.com/twpayne/chezmoi/assets/8222059/519c4166-66f8-4a0c-a927-80ea39708617)
Getting this error

**Reply by twpayne** [reply]
The error message is from the commit message template, not the `run_onchange_` script.

**Reply by rubiin** [reply]
I do not have any commit msg template 

![image](https://github.com/twpayne/chezmoi/assets/8222059/397b1d75-172c-4387-987e-0d776c692547)

**Reply by twpayne** [reply]
You are using the default commit message template.

Please take some time to investigate the causes of the errors you are seeing and demonstrate some effort in your requests for help here. I encourage you to read [How To Ask Questions The Smart Way](http://www.catb.org/esr/faqs/smart-questions.html). If you are still having trouble a week from now, feel free to ask for help again.
